src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/scripted_fields_table/components/table/table.tsx

```diff
@@ -19,34 +19,39 @@
 
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { EuiInMemoryTable, EuiBasicTableColumn } from '@elastic/eui';
 
-import { EuiInMemoryTable } from '@elastic/eui';
+import { ScriptedFieldItem } from '../../types';
+import { IIndexPattern } from '../../../../../../../../../../../plugins/data/public';
 
-import { i18n } from '@kbn/i18n';
+interface TableProps {
+  indexPattern: IIndexPattern;
+  items: ScriptedFieldItem[];
+  editField: (field: ScriptedFieldItem) => void;
+  deleteField: (field: ScriptedFieldItem) => void;
+}
 
-export class Table extends PureComponent {
+export class Table extends PureComponent<TableProps> {
```

Comment:
Lets name this `ScriptedFieldsTable` so the name is less generic.

Reply:
It's a great suggestion but we already have ScriptedFieldsTable component (it's a parent of Table)
Also at least 3 other components have the same file structure. Not sure that we should rename it now. But I understand your concern and it will be fixed when we finish migration other components. Probably we can unify something